### PR TITLE
Allow access TestingEngine to internal

### PR DESCRIPTION
**Pull request by shargon**
I need this change if I want to import from json using `SnapshotCache` in https://github.com/neo-project/neo-devpack-dotnet/pull/890

### CODE REVIEW

File: src/Neo/SmartContract/StorageKey.cs
Comment (cschuchardt88) [inline src/Neo/SmartContract/StorageKey.cs:37]: **DO NOT** expose the constructor. Their is a better way.

Add this to the `neo.csproj` file to expose the `internal` methods to the different assemblies. If you make this `public` then other developers (non-core devs) will start using it. Than we will have to support this interface.
```xml
<ItemGroup>
  <InternalsVisibleTo Include="Neo.Compiler.CSharp" />
  <InternalsVisibleTo Include="Neo.SmartContract.Testing" />
  <InternalsVisibleTo Include="Neo.SmartContract.Testing.UnitTests" />
</ItemGroup>
```

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]

File: src/Neo/SmartContract/StorageKey.cs
Comment (cschuchardt88) [reply src/Neo/SmartContract/StorageKey.cs:37]: I understand, and the xml code that I posted. Allows those `3` assemblies to use the `internal` methods. You will have access to `ALL` the `internal` methods in `neo.dll`.

Comment (shargon) [reply src/Neo/SmartContract/StorageKey.cs]: Both solutions are good for me, changed

**Review by Jim8y — APPROVED** [review]

File: src/Neo/SmartContract/StorageKey.cs
Comment (cschuchardt88) [reply src/Neo/SmartContract/StorageKey.cs]: Just make sure the fully qualified assembly name is correctly spelled. That is what `include` equals.

**Comment by shargon** [comment]
Wait for merge, I need to do some changes (I want to allow to mock specific calls)

**Comment by Jim8y** [comment]
run `git pull origin master` under the neo folder of the devpack project, you can get the new `neo`

**Pull request merged**
